packages/atomic/src/components/commerce/atomic-commerce-interface/e2e/atomic-commerce-interface.e2e.ts

```diff
@@ -0,0 +1,147 @@
+    }) => {
+      await page.goto(
+        'http://localhost:4400/iframe.html?id=atomic-commerce-interface--with-product-list&viewMode=story&args=attributes-language:foo'
+      );
+      await expect(commerceInterface.interface()).toContainText('Products');
+    });
+
+    test('should default back to the non region locale (e.g., "es-ES" to "es")', async ({
+      page,
+      commerceInterface,
+    }) => {
+      await page.goto(
+        'http://localhost:4400/iframe.html?id=atomic-commerce-interface--with-product-list&viewMode=story&args=attributes-language:es-ES'
+      );
+      await expect(commerceInterface.interface()).toContainText('Productos');
+    });
+
+    test('should support changing the language of the interface after initialization', async ({
+      page,
+      commerceInterface,
+    }) => {
+      await commerceInterface.hydrated.waitFor();
+
+      await page.evaluate(() => {
+        const commerceInterfaceComponent = document.querySelector(
+          'atomic-commerce-interface'
+        ) as HTMLAtomicCommerceInterfaceElement;
+
+        commerceInterfaceComponent.language = 'fr';
+      });
+
+      await expect(commerceInterface.interface()).toContainText('Produits');
+    });
+    test('should revert to english after changing the language to an invalid value', async ({
+      page,
+      commerceInterface,
+    }) => {
+      await page.goto(
+        'http://localhost:4400/iframe.html?id=atomic-commerce-interface--with-product-list&viewMode=story&args=attributes-language:fr'
+      );
+      await commerceInterface.hydrated.waitFor();
+
+      await page.evaluate(() => {
+        const commerceInterfaceComponent = document.querySelector(
+          'atomic-commerce-interface'
+        ) as HTMLAtomicCommerceInterfaceElement;
+
+        commerceInterfaceComponent.language = 'foo';
+      });
+
+      await expect(commerceInterface.interface()).toContainText('Products');
+    });
+  });
+
+  test.describe('when selecting a facet value', () => {
+    test.describe('when reflectStateInUrl is enabled', () => {
+      test('should update the url', async ({page, commerceInterface}) => {
+        const facetValueLabel = commerceInterface.getFacetValue('Nike');
+
+        await facetValueLabel.click();
+
+        await page.waitForURL(
+          '**/iframe.html?id=atomic-commerce-interface--with-product-list*'
+        );
+
+        const currentUrl = page.url();
+
+        expect(currentUrl).toContain('Nike');
+      });
+    });
+    test.describe('when reflectStateInUrl is not', () => {
+      test('should not update the url', async ({page, commerceInterface}) => {
+        await page.goto(
+          'http://localhost:4400/iframe.html?id=atomic-commerce-interface--with-product-list&viewMode=story&args=attributes-reflect-state-in-url:false'
+        );
+
+        const facetValueLabel = commerceInterface.getFacetValue('Nike');
+
+        await facetValueLabel.click();
+
+        await page.waitForTimeout(1000);
+
```

Comment:
Try to wait on 'something' instead than an arbitrary timeout.
For example, you could wait for something to appear in the page (I know, it'll be easier when we have a stable org/source, but still)